test/jdk/java/util/logging/FileHandlerAccessTest.java

```diff
@@ -0,0 +1,110 @@
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.logging.FileHandler;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+
+/**
+ * @test
+ * @bug 8252883
+ * @summary tests the handling of AccessDeniedException due to delay in Windows deletion.
+ * @modules java.logging/java.util.logging:open
+ * @run main/othervm FileHandlerAccessTest process 20
+ * @run main/othervm FileHandlerAccessTest thread 20
+ * @author evwhelan
+ */
+
+public class FileHandlerAccessTest {
+    public static void main(String[] args) throws Exception {
+        if (!(args.length == 2 || args.length == 1)) {
+            System.out.println("Usage error: expects java FileHandlerAccessTest [process/thread] <count>");
+            return;
+        }
+        else if (args.length == 2) {
+            var type = args[0];
+            var count = Integer.parseInt(args[1]);
+
+            for (var i = 0; i < count; i++) {
+                System.out.println("Testing with arguments: type=" + type + ", count="+count);
+                if (type.equals("process")) {
+                    new Thread(FileHandlerAccessTest::accessProcess).start();
+                }
+                else if (type.equals("thread")) {
+                    new Thread(FileHandlerAccessTest::access).start();
+                }
+            }
+        }
+        else {
+            access();
+        }
+    }
+
+    private static void access() {
+        try {
+            var handler = new FileHandler("sample%g.log", 1048576, 2, true);
+            handler.publish(new LogRecord(Level.SEVERE, "TEST"));
+            handler.close();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
```

Comment:
Will throwing an exception in a thread (if happening in the Thread created at line 60) cause jtreg to fail? Or will the exception simply be discarded?

Reply:
The exception will in fact make the jtreg test fail. I verified this by creating a dummy test case and throwing an exception within a thread as you suggested.